sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_intel_matrix.asciidoc

```diff
@@ -203,6 +203,106 @@ In the case of `ext_intel_packed` matrix memory layout, `row` and
 `col` represent the coordinates in the logical matrix before VNNI
 transformation.
 
+=== Load/Store/Fill With Out-of-Bounds Checks
+`joint_matrix_load`, `joint_matrix_store`, and `joint_matrix_fill`
+operations do not perform bounds checking. When the global matrix size is
+not multiple of the joint matrix size, this extension adds a new form
+of API to load, store, and fill joint matrix while checking the
+bounds. For load and fill, the out-of-bounds elements are set to
+0. For the store, they are dropped out.
+
+These functions are similar to the existing ones without bounds
+checking, namely `joint_matrix_fill`, `joint_matrix_load`, and
+`joint_matrix_store`. But they are different in three ways:
+
+* The pointer `src` designates the base pointer of the original memory
+matrix excluding any offset calculations. The coordinates are given
+separately here.
+
+* The coordinates to calculate the base pointer offset to load/store are
+given as separate arguments.  
+
+* These variants take extra arguments to determine the global bounds
+`Height` and `Width` of the two-dimensional region to load to or store
+from.
+
+
+The two-dimensional memory region is interpreted as follows, where
+`Height` is number of rows in the memory matrix, `Width` is number of
+columns in the memory matrix, `stride` is number of rows that include
+the out of bounds data (depicted as x here).
+
+|<-------------`Width`---------->|
+
+|DDDDDDDDDDDDDDDDDxxx|
+
+|DDDDDDDDDDDDDDDDDxxx|
+
+|......................................................| `Height`
+
+|DDDDDDDDDDDDDDDDDxxx|
+
+|DDDDDDDDDDDDDDDDDxxx|
+
+|<------------`Stride`---------->|
+ 
+Note that joint matrix shape `Rows` and `Cols` represents a sub-block
+of the picture above for each sub-group execution. At the last
+iteration of the range of the launched kernel, the remainder that
+should exclude the out-of-bounds data is checked against `Width` and
+`Height`. For instance, load of `sub_b` defined as
+`joint_matrix<sub_group, bfloat16, use::b, 2, 3, layout::row_major> sub_b;`
+checks that `Height=10` is not multiple of `3` and the data (2
+elements left) that exceeds getting the total `10` rows will be filled
+by zero.
+
+```c++
+namespace sycl::ext::intel::experimental::matrix {
+
```

Comment:
@fda0 does this change address the missing arguments to generate efficiently 2d load/store in IGC?

Reply:
Yes I changed it to add the two offset coordinates separately and excluded them from the base pointer calculation.